journal/src/main/java/io/camunda/zeebe/journal/file/JournalSegment.java

```diff
@@ -145,8 +145,11 @@ public MappedJournalSegmentWriter writer() {
    */
   MappedJournalSegmentReader createReader() {
     checkOpen();
-    return new MappedJournalSegmentReader(
-        buffer.asReadOnlyBuffer().position(0).order(ENDIANNESS), this, index);
+    final MappedJournalSegmentReader reader =
+        new MappedJournalSegmentReader(
+            buffer.asReadOnlyBuffer().position(0).order(ENDIANNESS), this, index);
+    readers.add(reader);
```

Comment:
:facepalm: Readers were not tracked :smile:

Reply:
:smile: Good catch! I assumed they were, didn't think of checking :sweat_smile: